jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/internal/EthereumSignatureVerifier.java

```diff
@@ -0,0 +1,57 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe.internal;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import org.bouncycastle.jcajce.provider.digest.Keccak;
+import org.eclipse.jetty.util.StringUtil;
+import org.web3j.crypto.ECDSASignature;
+import org.web3j.crypto.Keys;
+import org.web3j.crypto.Sign;
+
+public class EthereumSignatureVerifier
```

Comment:
javadoc

Reply:
javadoc with links/references to the algorithm specification and background

Reply:
done